public/app/features/scenes/variables/sets/SceneVariableSet.ts

```diff
@@ -0,0 +1,157 @@
+import { Unsubscribable } from 'rxjs';
+
+import { SceneObjectBase } from '../../core/SceneObjectBase';
+import { SceneVariable, SceneVariables, SceneVariableSetState, SceneVariableValueChangedEvent } from '../types';
+
+export class SceneVariableSet extends SceneObjectBase<SceneVariableSetState> implements SceneVariables {
+  /** Variables that have changed in since the activation or since the first manual value change */
+  // private variablesThatHaveChanged = new Map<string, SceneVariable>();
+
+  /** Variables that are scheduled to be validated and updated */
+  private variablesToUpdate = new Map<string, SceneVariable>();
+
+  /** Cached variable dependencies */
+  private dependencies = new Map<string, string[]>();
+
+  /** Variables currently updating  */
+  private updating = new Map<string, VariableUpdateInProgress>();
```

Comment:
Somehow I liked the previous approach with the UpdateManager - but no strong stand on that. Any particular constraint that you hit that made you move the logic into the object?

Reply:
mainly to reduce the number of components involved, and code wise this pretty small so does not need to broken up (yet), the tests where written for the SceneVariableSet, etc, it had the subscriptions etc. It just became a bit easier to follow the logic

Reply:
When I added the deactivate and cleanup logic it just became to much of a pointless delegation to the UpdateManager